app/index.js

```diff
@@ -3,6 +3,14 @@ const messageService = require('./messaging')
 const paymentProcessing = require('./processing')
 const batching = require('./batching')
 
+const createServer = require('./server')
+const init = async () => {
+  const server = await createServer()
+  await server.start()
+  console.log('Server running on %s', server.info.uri)
+}
+init()
```

Comment:
I'd maybe move this down to the module.exports method below.  It's a promise but it's not awaited and isnt' started consistently with the other processes.

What about removing this line and instead changing the `startService` function to be:

```
module.exports = (async function startService () {
  await init()
  await messageService.start()
  await paymentProcessing.start()
  await batching.start()
}())
```

Could probably also debate that should be a promise all scenario, but will leave it to your discretion.